lib/features/wordcloud/config.dart

```diff
@@ -300,13 +304,27 @@ class _ConfigState extends ConsumerState<WordCloudConfig> {
     );
   }
 
+  String sanitiseMaxWord(String txt) {
+    // it should be int or Inf. Otherwise, convert to an Inf.
+    if (txt != 'Inf') {
+      int? maxWord = int.tryParse(txt);
+      if (maxWord == null) {
+        return 'Inf';
+      }
+      return maxWord.toString();
+    }
+    return txt;
+  }
+
   void _updateMaxWordProvider() {
     debugPrint('max word text changed to ${maxWordTextController.text}');
-    ref.read(maxWordProvider.notifier).state = maxWordTextController.text;
+    ref.read(maxWordProvider.notifier).state =
+        sanitiseMaxWord(maxWordTextController.text);
   }
 
   void _updateMinFreqProvider() {
     debugPrint('min freq text changed to ${minFreqTextController.text}');
-    ref.read(minFreqProvider.notifier).state = minFreqTextController.text;
+    ref.read(minFreqProvider.notifier).state =
```

Comment:
If you reset it to 1, will the UI reflect that?

Reply:
yes